### PR TITLE
make sure all log messages are safe

### PR DESCRIPTION
**Pull request by pnepywoda**

### CODE REVIEW

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RetryInterceptor.java [hunk 1]
```diff
@@ -60,7 +61,7 @@ public Response intercept(Chain chain) throws IOException {
                 }
             }
             if (!successful) {
-                log.error("Request to url {} failed on attempt {}", request.url(), numTries);
+                log.error("Request to url {} failed on attempt {}", request.url(), SafeArg.of("numTries", numTries));
```
Comment (markelliot) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RetryInterceptor.java]: log message should get cleaned up to:

    "Failed request to url"

File: tracing/src/main/java/com/palantir/remoting3/tracing/AsyncSpanObserver.java [hunk 2]
```diff
@@ -75,7 +76,7 @@ public void onFailure(Throwable error) {
             });
         } else {
             log.trace("Failed to notify span observer since the maximum number of allowed concurrent observations was "
-                    + "exceeded: {}", maxInflights);
+                    + "exceeded: {}", SafeArg.of("maxInflights", maxInflights));
```
Comment (markelliot) [inline tracing/src/main/java/com/palantir/remoting3/tracing/AsyncSpanObserver.java]: Log message should get cleaned up to remove braces.

**Review by markelliot — CHANGES_REQUESTED** [review]

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RetryInterceptor.java (hunk 1 above)
Comment (pnepywoda) [reply okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RetryInterceptor.java]: why? don't like the unsafe url being logged?

File: tracing/src/main/java/com/palantir/remoting3/tracing/AsyncSpanObserver.java (hunk 2 above)
Comment (pnepywoda) [reply tracing/src/main/java/com/palantir/remoting3/tracing/AsyncSpanObserver.java]: why? it'll log the safe arg there

Comment (markelliot) [reply tracing/src/main/java/com/palantir/remoting3/tracing/AsyncSpanObserver.java]: per the spec, the substitution never happens/the args are named and shown along side.

**Review by markelliot — APPROVED** [review]

**Review by uschi2000 — APPROVED** [review]

**Comment by uschi2000** [comment]
try `ag '{}'`, there's still a few bad ones

**Comment by pnepywoda** [comment]
Sure but they'll be unsafe args

**Pull request merged**